Library/Homebrew/extend/os/linux/dev-cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,245 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "formula"
+
+class TestRunnerFormula
+  extend T::Sig
+
+  sig { returns(String) }
+  attr_reader :name
+
+  sig { returns(Formula) }
+  attr_reader :formula
+
+  sig { params(name: String).void }
+  def initialize(name)
+    @name = T.let(name, String)
+    @formula = T.let(Formula[name], Formula)
+    @dependent_hash = T.let({}, T::Hash[T::Boolean, T::Array[TestRunnerFormula]])
+    freeze
+  end
+
+  sig { returns(T::Boolean) }
+  def macos_only?
+    formula.requirements.any? { |r| r.is_a?(MacOSRequirement) && !r.version_specified? }
+  end
+
+  sig { returns(T::Boolean) }
+  def linux_only?
+    formula.requirements.any?(LinuxRequirement)
+  end
+
+  sig { returns(T::Boolean) }
+  def x86_64_only?
+    formula.requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :x86_64) }
+  end
+
+  sig { returns(T::Boolean) }
+  def arm64_only?
+    formula.requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :arm64) }
+  end
+
+  sig { returns(T.nilable(MacOSRequirement)) }
+  def versioned_macos_requirement
+    formula.requirements.find { |r| r.is_a?(MacOSRequirement) && r.version_specified? }
+  end
+
+  sig { params(macos_version: MacOS::Version).returns(T::Boolean) }
+  def compatible_with?(macos_version)
+    # Assign to a variable to assist type-checking.
+    requirement = versioned_macos_requirement
+    return true if requirement.blank?
+
+    macos_version.public_send(requirement.comparator, requirement.version)
+  end
+
+  sig { returns(T::Array[TestRunnerFormula]) }
+  def dependents
+    @dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?] ||= with_env(HOMEBREW_STDERR: "1") do
+      Utils.safe_popen_read(
+        HOMEBREW_BREW_FILE, "uses", "--formulae", "--eval-all", "--include-build", "--include-test", name
+      ).split("\n").map { |dependent| TestRunnerFormula.new(dependent) }.freeze
+    end
+
+    T.must(@dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?])
+  end
+end
+
+module Homebrew
+  extend T::Sig
+
+  sig {
+    params(
+      testing_formulae:     T::Array[TestRunnerFormula],
+      reject_platform:      T.nilable(Symbol),
+      reject_arch:          T.nilable(Symbol),
+      select_macos_version: T.nilable(MacOS::Version),
+    ).returns(T::Boolean)
+  }
+  def self.formulae_have_untested_dependents?(testing_formulae, reject_platform:, reject_arch:, select_macos_version:)
+    testing_formulae.any? do |formula|
+      # If the formula has a platform/arch/macOS version requirement, then its
+      # dependents don't need to be tested if these requirements are not satisfied.
+      next false if reject_platform && formula.method("#{reject_platform}_only?".to_sym).call
+      next false if reject_arch && formula.method("#{reject_arch}_only?".to_sym).call
```

Comment:
```suggestion
      next false if reject_platform && formula.method(:"#{reject_platform}_only?").call
      next false if reject_arch && formula.method(:"#{reject_arch}_only?").call
```

And similarly below. (Very minor suggestion, feel free to ignore if it's an intentional choice)

Reply:
Not intentional at all! I just reached for the first thing that came to mind to try to appease the type-checker.